addons/mail/wizard/mail_compose_message.py

```diff
@@ -512,17 +512,26 @@ def _process_state(self, mail_values_dict):
 
         return mail_values_dict
 
-    def _get_blacklist_record_ids(self, mail_values_dict):
+    def _get_blacklist_record_ids(self, mail_values_dict, recipients_info):
         blacklisted_rec_ids = set()
-        if self.composition_mode == 'mass_mail' and issubclass(type(self.env[self.model]), self.pool['mail.thread.blacklist']):
-            self.env['mail.blacklist'].flush_model(['email', 'active'])
+        if self.composition_mode == 'mass_mail':
+            self.env['mail.blacklist'].flush(['email'])
             self._cr.execute("SELECT email FROM mail_blacklist WHERE active=true")
             blacklist = {x[0] for x in self._cr.fetchall()}
             if blacklist:
-                targets = self.env[self.model].browse(mail_values_dict.keys()).read(['email_normalized'])
-                # First extract email from recipient before comparing with blacklist
-                blacklisted_rec_ids.update(target['id'] for target in targets
-                                           if target['email_normalized'] in blacklist)
+                if issubclass(type(self.env[self.model]), self.pool['mail.thread.blacklist']):
+                    targets = self.env[self.model].browse(mail_values_dict.keys()).read(['email_normalized'])
+                    # First extract email from recipient before comparing with blacklist
+                    blacklisted_rec_ids.update(target['id'] for target in targets
+                                               if target['email_normalized'] in blacklist)
+                else:
```

Comment:
please apply the stable comments here
(or even better, apply the change on the stable branch and then re-cherry pick the commit here)